pkg/model/awsmodel/api_loadbalancer.go

```diff
@@ -418,35 +426,44 @@ func (b *APILoadBalancerBuilder) Build(c *fi.ModelBuilderContext) error {
 						SecurityGroup: masterGroup.Task,
 						ToPort:        fi.Int64(443),
 					}
-					if utils.IsIPv6CIDR(cidr) {
-						t.IPv6CIDR = fi.String(cidr)
-					} else {
-						t.CIDR = fi.String(cidr)
-					}
+					t.SetCidrOrPrefix(cidr)
 					AddDirectionalGroupRule(c, t)
 				}
 
+				icmpIpv6 := false
+				if strings.HasPrefix(cidr, "pl-") {
+					b, err := b.IsIPv6PrefixList(cidr)
+					if err != nil {
+						return err
+					}
+					icmpIpv6 = b
+				} else {
+					icmpIpv6 = utils.IsIPv6CIDR(cidr)
+				}
```

Comment:
I don't think model should check the kind of prefix with AWS. I would rather skip the ICMP rules for prefixes, for now.

As a curiosity, any idea what happens if you set IPv4 PrefixList to ICMPv6 protocol or IPV6 PrefixList to ICMPv4 protocol?

Reply:
Just tried that out, AWS does allow to configure it like that, but not sure what the net effect is. I added a ICMPv4 inbound rule with an IPv6 type PrefixList, and the console accepted and created that.

Reply:
So, let's skip this for now, to get it into 1.23 before the release. We can try later to add the ICMP rules and see what happens.

Reply:
I'm not sure what the effect would be of not rendering ICMP rules when a prefix list is in use? I took the approach of querying the AWS API from the IAM model, where that was already used, so I figured that was allowed ;).

Are you sure we can skip them?

Reply:
It is required for [Path MTU Discovery](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/network_mtu.html#path_mtu_discovery).
I am sure I don't want to call the cloud for this. 😄